scripts/generate-llms-txt.js

```diff
@@ -0,0 +1,374 @@
+              } else if (nestedItem.type === 'doc' && nestedItem.id) {
+                const fullDocPath = `${docPath}${mapDocPath(nestedItem, prefix)}`;
+                const {title, slug} = extractMetadataFromMarkdown(fullDocPath);
+                markdown += `- [${title}](${URL_PREFIX}${prefix}/${slug ?? nestedItem.id})\n`;
+              }
+            });
+          }
+        }
+      });
+    });
+  });
+
+  // Format and cleanup whitespaces
+  return markdown.replace(/(#+ .*)\n/g, '\n$1\n').replace(/\n(\n)+/g, '\n\n');
+}
+
+const inputFilePaths = [
+  {
+    name: 'sidebars.ts',
+    docPath: '../docs/',
+    prefix: '/docs',
+  },
+  {
+    name: 'sidebarsArchitecture.ts',
+    docPath: './architecture/',
+    prefix: '/architecture',
+  },
+  {
+    name: 'sidebarsCommunity.ts',
+    docPath: './community/',
+    prefix: '/community',
+  },
+  {
+    name: 'sidebarsContributing.ts',
+    docPath: './contributing/',
+    prefix: '/contributing',
+  },
+];
+
+let output = `# ${TITLE}\n\n`;
+output += `> ${DESCRIPTION}\n\n`;
+output += `This documentation covers all aspects of using React Native, from installation to advanced usage.\n\n`;
+
+const generateOutput = () => {
+  const results = [];
+  const promises = [];
+
+  for (const {name, docPath, prefix} of inputFilePaths) {
+    const inputFilePath = `./${name}`;
+    const outputFilePath = inputFilePath.replace(/\.tsx?$/, '-urls.txt');
+
+    const sidebarConfig = convertSidebarConfigToJson(inputFilePath);
+    if (sidebarConfig) {
+      const urls = extractUrlsFromSidebar(sidebarConfig, prefix);
+
+      // First check URLs for 404 errors
+      const promise = processUrls(urls)
+        .then(result => {
+          if (result.unavailableUrls.length === 0) {
+            // Only generate documentation if all URLs are valid
+            const markdown = generateMarkdown(sidebarConfig, docPath, prefix);
+            results.push({ markdown, prefix });
+            console.log(`Successfully generated output from ${inputFilePath}`);
+          } else {
+            console.error(
+              'Documentation generation skipped due to broken links'
+            );
+            process.exit(1);
+          }
+        })
+        .catch(err => {
+          console.error('Error processing URLs:', err);
+          process.exit(1);
+        });
+
+      promises.push(promise);
+    } else {
+      console.error('Failed to convert sidebar config to JSON');
+      process.exit(1);
+    }
+  }
+
+  // Wait for all promises to complete before writing the file
+  Promise.all(promises)
+    .then(() => {
+      // Sort results to ensure docs section is first
+      results.sort((a, b) => {
+        if (a.prefix === '/docs') return -1;
+        if (b.prefix === '/docs') return 1;
+        return 0;
+      });
+
+      // Combine all markdown content in the correct order
+      output += results.map(r => r.markdown).join('\n');
+      
```

Comment:
```suggestion

```

ESLint locally wants to cleanup whitespace here, let's make sure the file is well formatted.